lib/Sema/ConstraintSystem.h

```diff
@@ -837,9 +837,21 @@ class Solution {
   llvm::DenseMap<std::pair<CanType, CanType>, ConversionRestrictionKind>
     ConstraintRestrictions;
 
+  using ConstraintFixVector = llvm::SmallVector<ConstraintFix *, 4>;
   /// The list of fixes that need to be applied to the initial expression
   /// to make the solution work.
-  llvm::SmallVector<ConstraintFix *, 4> Fixes;
+  ConstraintFixVector Fixes;
+
+  using AggregatedFixesMap = llvm::SmallMapVector<ConstraintLocator *,
+      llvm::SmallMapVector<FixKind, ConstraintFixVector, 4>, 4>;
+
+  AggregatedFixesMap getAggregatedFixes() const {
```

Comment:
This temporary aggregation doesn't seem essential, `diagnoseAmbiguityWithFixes` should be able to do this in one go...